packages/tree/src/tree.vue

```diff
@@ -0,0 +1,339 @@
+      :show-checkbox="showCheckbox"
+      :render-content="renderContent"
+      @node-expand="handleNodeExpand"
+    />
+    <div v-if="isEmpty" class="el-tree__empty-block">
+      <span class="el-tree__empty-text">{{ emptyText }}</span>
+    </div>
+    <div
+      v-show="dragState.showDropIndicator"
+      ref="dropIndicator$"
+      class="el-tree__drop-indicator"
+    >
+    </div>
+  </div>
+</template>
+<script lang='ts'>
+import { defineComponent, ref, provide, computed, watch, PropType, getCurrentInstance, ComponentInternalInstance } from 'vue'
+import TreeStore from './model/tree-store'
+import { getNodeKey as getNodeKeyUtil } from './model/util'
+import ElTreeNode from './tree-node.vue'
+import { useNodeExpandEventBroadcast } from './model/useNodeExpandEventBroadcast'
+import { useDragNodeHandler } from './model/useDragNode'
+import { useKeydown } from './model/useKeydown'
+import Node from './model/node'
+import { t } from '@element-plus/locale'
+import {
+  TreeComponentProps,
+  TreeNodeData,
+  TreeKey,
+  TreeData,
+  RootTreeType,
+} from './tree.d'
+
+export default defineComponent({
+  name: 'ElTree',
+  components: { ElTreeNode },
+  props: {
+    data: {
+      type: Array,
+    },
+    emptyText: {
+      type: String,
+      default() {
+        return t('el.tree.emptyText')
+      },
+    },
+    renderAfterExpand: {
+      type: Boolean,
+      default: true,
+    },
+    nodeKey: String,
+    checkStrictly: Boolean,
+    defaultExpandAll: Boolean,
+    expandOnClickNode: {
+      type: Boolean,
+      default: true,
+    },
+    checkOnClickNode: Boolean,
+    checkDescendants: {
+      type: Boolean,
+      default: false,
+    },
+    autoExpandParent: {
+      type: Boolean,
+      default: true,
+    },
+    defaultCheckedKeys: Array,
+    defaultExpandedKeys: Array,
+    currentNodeKey: [String, Number] as PropType<string | number>,
+    renderContent: Function,
+    showCheckbox: {
+      type: Boolean,
+      default: false,
+    },
+    draggable: {
+      type: Boolean,
+      default: false,
+    },
+    allowDrag: Function,
+    allowDrop: Function,
+    props: {
+      type: Object,
+      default() {
+        return {
+          children: 'children',
+          label: 'label',
+          disabled: 'disabled',
+        }
+      },
+    },
+    lazy: {
+      type: Boolean,
+      default: false,
+    },
+    highlightCurrent: Boolean,
+    load: Function,
+    filterNodeMethod: Function,
+    accordion: Boolean,
+    indent: {
+      type: Number,
+      default: 18,
+    },
+    iconClass: String,
+  },
+  emits: [
+    'check-change',
+    'current-change',
+    'node-click',
+    'node-contextmenu',
+    'node-collapse',
+    'node-expand',
+    'check',
+    'node-drag-start',
+    'node-drag-end',
+    'node-drop',
+    'node-drag-leave',
+    'node-drag-enter',
+    'node-drag-over',
+  ],
+  setup(props: TreeComponentProps, ctx) {
+
+    const store = ref<TreeStore>(new TreeStore({
```

Comment:
Is this ref necessary? I didn't see any place we will update store.

Reply:
necessary, because of store have some methods will update it self, if store is not ref, store will not be reactively